tests/contrib/vertica/test_vertica.py

```diff
@@ -0,0 +1,341 @@
+        assert not isinstance(
+            vertica_python.vertica.cursor.Cursor.execute, wrapt.ObjectProxy
+        )
+
+
+class TestVertica(object):
+    # def setup_method(self, method):
+    #     patch()
+
+    def teardown_method(self, method):
+        unpatch()
+
+    @override_config({"service_name": "test_svc_name"})
+    def test_configuration_service_name(self):
+        """Ensure that the integration can be configured."""
+        patch()
+        import vertica_python
+
+        test_tracer = get_dummy_tracer()
+
+        conn = vertica_python.connect(**VERTICA_CONFIG)
+        cur = conn.cursor()
+        Pin.override(cur, tracer=test_tracer)
+        cur.execute("DROP TABLE IF EXISTS {}".format(TEST_TABLE))
+        conn.close()
+        spans = test_tracer.writer.pop()
+        assert len(spans) == 1
+        assert spans[0].service == "test_svc_name"
+
+    @override_config(
+        {
+            "patch": {
+                "vertica_python.vertica.connection.Connection": {
+                    "routines": {
+                        "cursor": {
+                            "operation_name": "get_cursor",
+                            "trace_enabled": True,
+                        }
+                    }
+                }
+            }
+        }
+    )
+    def test_configuration_routine(self):
+        """Ensure that the integration routines can be configured."""
+        patch()
+        import vertica_python
+
+        test_tracer = get_dummy_tracer()
+
+        conn = vertica_python.connect(**VERTICA_CONFIG)
+        Pin.override(conn, tracer=test_tracer)
+        conn.cursor()  # should be traced now
+        conn.close()
+        spans = test_tracer.writer.pop()
+        assert len(spans) == 1
+        assert spans[0].name == "get_cursor"
+
+    def test_execute_metadata(self, test_conn, test_tracer):
+        """Metadata related to an `execute` call should be captured."""
+        conn, cur = test_conn
+
+        Pin.override(cur, tracer=test_tracer)
+
+        cur.execute("INSERT INTO {} (a, b) VALUES (1, 'aa');".format(TEST_TABLE))
+        cur.execute("SELECT * FROM {};".format(TEST_TABLE))
+        conn.close()
+
+        spans = test_tracer.writer.pop()
+        assert len(spans) == 2
+
+        # check all the metadata
+        assert spans[0].service == "vertica"
+        assert spans[0].span_type == "sql"
+        assert spans[0].name == "vertica.query"
+        assert spans[0].get_metric("db.rowcount") == -1
+        query = "INSERT INTO test_table (a, b) VALUES (1, 'aa');"
+        assert spans[0].get_tag("query") == query
+        assert spans[0].get_tag("out.host") == "127.0.0.1"
+        assert spans[0].get_tag("out.port") == "5433"
+        assert spans[0].get_tag("db.name") == "docker"
+        assert spans[0].get_tag("db.user") == "dbadmin"
+
+        assert spans[1].get_tag("query") == "SELECT * FROM test_table;"
+
+    def test_cursor_override(self, test_conn):
```

Comment:
This test doesn't seem much different from the last, since both are setting `(cur, tracer=test_tracer)` it is just whether it is the `pytest` fixture `test_tracer` or one you are creating right now.

No strong opinion on whether this test should stay/go, but seems unnecessary.

Reply:
This test should be the only one using `Pin.override(cur, tracer=test_tracer)` to ensure that a user can override the tracer used if they desire. However, it seems like I copied it across for some of the other tests. So I'll clean those up.